programs/uxd/src/instructions/credix_lp/rebalance_request_create_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,284 @@
+    pub depository_shares: Box<Account<'info, TokenAccount>>,
+
+    /// #8
+    #[account()]
+    pub credix_global_market_state: Box<Account<'info, credix_client::GlobalMarketState>>,
+
+    /// #9
+    /// CHECK: unused by us, checked by credix
+    pub credix_signing_authority: AccountInfo<'info>,
+
+    /// #10
+    #[account(mut)]
+    pub credix_liquidity_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #11
+    #[account(mut)]
+    pub credix_shares_mint: Box<Account<'info, Mint>>,
+
+    /// #12
+    #[account(
+        owner = credix_client::ID,
+        seeds = [
+            credix_global_market_state.key().as_ref(),
+            depository.key().as_ref(),
+            CREDIX_LP_EXTERNAL_PASS_NAMESPACE
+        ],
+        bump,
+        seeds::program = credix_client::ID,
+        constraint = credix_pass.user == depository.key() @UxdError::InvalidCredixPass,
+        constraint = credix_pass.disable_withdrawal_fee @UxdError::InvalidCredixPassNoFees,
+    )]
+    pub credix_pass: Account<'info, credix_client::CredixPass>,
+
+    /// #13
+    #[account(
+        mut,
+        owner = credix_client::ID,
+        seeds = [
+            credix_global_market_state.key().as_ref(),
+            &credix_global_market_state.latest_withdraw_epoch_idx.to_le_bytes(),
+            CREDIX_LP_EXTERNAL_WITHDRAW_EPOCH_NAMESPACE
+        ],
+        bump,
+        seeds::program = credix_client::ID,
+    )]
+    pub credix_withdraw_epoch: Account<'info, credix_client::WithdrawEpoch>,
+
+    /// #14
+    /// CHECK: initialized by credix program, not manipulated by us
+    #[account(mut)]
+    pub credix_withdraw_request: AccountInfo<'info>,
+
+    /// #15
+    pub system_program: Program<'info, System>,
+    /// #16
+    pub credix_program: Program<'info, credix_client::program::Credix>,
+}
+
+pub(crate) fn handler(ctx: Context<RebalanceRequestCreateFromCredixLpDepository>) -> Result<()> {
+    let credix_global_market_state = ctx.accounts.depository.load()?.credix_global_market_state;
+    let collateral_mint = ctx.accounts.depository.load()?.collateral_mint;
+    let depository_pda_signer: &[&[&[u8]]] = &[&[
+        CREDIX_LP_DEPOSITORY_NAMESPACE,
+        credix_global_market_state.as_ref(),
+        collateral_mint.as_ref(),
+        &[ctx.accounts.depository.load()?.bump],
+    ]];
+
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Check if the withdraw request period is active at the moment
+    // ---------------------------------------------------------------------
+
+    let start_of_request_phase_timestamp = ctx.accounts.credix_withdraw_epoch.go_live;
+    let end_of_request_phase_timestamp = start_of_request_phase_timestamp
+        .checked_add(ctx.accounts.credix_withdraw_epoch.request_seconds.into())
+        .ok_or(UxdError::MathError)?;
+    let current_unix_timestamp = Clock::get()?.unix_timestamp;
+    if current_unix_timestamp < start_of_request_phase_timestamp
```

Comment:
`require!((start_of_request_phase_timestamp.. end_of_request_phase_timestamp).contains(current_unix_timestamp), Err(UxdError::InvalidCredixWithdrawEpochRequestPeriod.into()));`

nit, should work too

Reply:
Sounds good! applying now